src/PuJetIdSF.cc

```diff
@@ -134,16 +142,69 @@ float PuJetIdSF::getSF(bool isReal, float pt, float eta)
   return sf;
 }
 
+// getSFError
+float PuJetIdSF::getSFError(bool isReal, float pt, float eta)
+{
+  float sferr;
+
+  if (pt<20.) pt=20.;
+  if (pt>50.) pt=50.;
+
+  if (isReal)
+  {
+    int binglobal = h_eff_SF_err_->FindBin (pt, eta);
+    int binx, biny, binz;
+    h_eff_SF_err_->GetBinXYZ (binglobal, binx, biny, binz); // converts to x, y bins
+    int nx = h_eff_SF_err_->GetNbinsX();
+    int ny = h_eff_SF_err_->GetNbinsY();
 
-//getEvtWeight
-float PuJetIdSF::getEvtWeight(bigTree &theBigTree, TLorentzVector tau1, TLorentzVector tau2)
+    // under-overflows
+    if (binx < 1) binx = 1;
+    if (biny < 1) biny = 1;
+    if (binx > nx) binx = nx;
+    if (biny > ny) biny = ny;
+
+    sferr = h_eff_SF_err_->GetBinContent (binx, biny);
+  }
+  else
+  {
+    if (fabs(eta) > 2.5)
+    {
+      int binglobal = h_mistag_SF_err_->FindBin (pt, eta);
+      int binx, biny, binz;
+      h_mistag_SF_err_->GetBinXYZ (binglobal, binx, biny, binz); // converts to x, y bins
+      int nx = h_mistag_SF_err_->GetNbinsX();
+      int ny = h_mistag_SF_err_->GetNbinsY();
+
+      // under-overflows
+      if (binx < 1) binx = 1;
+      if (biny < 1) biny = 1;
+      if (binx > nx) binx = nx;
+      if (biny > ny) biny = ny;
+
+      sferr = h_mistag_SF_->GetBinContent (binx, biny);
+    }
+    else /* fabs(eta) < 2.5 */
+    {
+      // inside tracker acceptance use the SF itself as error on the mistag
+      sferr = getSF(isReal, pt, eta);
+    }
+  }
+
+  return sferr;
+}
+
```

Comment:
If I understood correctly this method retrieves the error on the mistake rate. If so you need a one for the efficiency as well.

Reply:
as I said in https://github.com/LLRCMS/KLUBAnalysis/pull/176#discussion_r504449055 the systematics are onòy provided for the SF, so the SF is shifted up/down, while the efficiency always is the same.